File: src/main/java/com/lmp/auth/dto/UserResponse.java
package com.lmp.auth.dto;

import com.lmp.auth.domain.User;

import java.time.LocalDateTime;
import java.util.Set;
import java.util.UUID;
import java.util.stream.Collectors;

/**
 * DTO de réponse pour les données utilisateur (sans mot de passe).
 */
public record UserResponse(
        UUID id,
        String email,
        String firstName,
        String lastName,
        String displayName,
        String phone,
        String city,
        String country,
        String companyName,
        String status,
        boolean emailVerified,
        boolean accountLocked,
        Set<String> roles,
        LocalDateTime registrationDate,
        LocalDateTime lastLoginDate,
        boolean vatReverseCharge,
        String vatNumber
) {
    public static UserResponse from(User user) {
        Set<String> roleNames = user.getRoles() != null
                ? user.getRoles().stream().map(r -> r.getName()).collect(Collectors.toSet())
                : Set.of();

        return new UserResponse(
                user.getId(),
                user.getEmail(),
                user.getFirstName(),
                user.getLastName(),
                user.getDisplayName(),
                user.getPhone(),
                user.getCity(),
                user.getCountry(),
                user.getCompanyName(),
                user.getStatus() != null ? user.getStatus().name() : "ACTIVE",
                Boolean.TRUE.equals(user.getEmailVerified()),
                Boolean.TRUE.equals(user.getAccountLocked()),
                roleNames,
                user.getRegistrationDate(),
                user.getLastLoginDate(),
                Boolean.TRUE.equals(user.getVatReverseCharge()),
                user.getVatNumber());
    }
}

File: backend-dotnet/src/Lmp.Application/Auth/UserResponse.cs
using Lmp.Domain.Auth;

namespace Lmp.Application.Auth;

/// <summary>User data without the password. Faithful port of <c>com.lmp.auth.dto.UserResponse</c>.</summary>
public sealed record UserResponse(
    Guid Id,
    string? Email,
    string? Username,
    string? FirstName,
    string? LastName,
    string DisplayName,
    string? Phone,
    string? City,
    string? Country,
    string? CompanyName,
    string Status,
    bool EmailVerified,
    bool AccountLocked,
    IReadOnlySet<string> Roles,
    DateTime RegistrationDate,
    DateTime? LastLoginDate,
    bool VatReverseCharge,
    string? VatNumber)
{
    public static UserResponse From(User user)
    {
        var roleNames = user.Roles.Select(r => r.Name).ToHashSet(StringComparer.Ordinal);
        return new UserResponse(
            user.Id,
            user.Email,
            user.Username,
            user.FirstName,
            user.LastName,
            user.GetDisplayName(),
            user.Phone,
            user.City,
            user.Country,
            user.CompanyName,
            StatusName(user.Status),
            user.EmailVerified,
            user.AccountLocked,
            roleNames,
            user.RegistrationDate,
            user.LastLoginDate,
            user.VatReverseCharge,
            user.VatNumber);
    }

    private static string StatusName(UserStatus status) => status switch
    {
        UserStatus.Active => "ACTIVE",
        UserStatus.Inactive => "INACTIVE",
        UserStatus.Deleted => "DELETED",
        _ => "ACTIVE",
    };
}
